### PR TITLE
Windows certificate import improvements

### PR DESCRIPTION
**Pull request by bretambrose**
* Check and log all errors.  Previously certain system calls could fail and this would not be noticed.
* Refactor windows cert import to follow newer standards for RAII and cleanup
* Adds support for Windows ECC key import.  Requires aws-c-cal to link to NCrypt.

Owes a tremendous amount to the Azure Windows key import examples found here: https://github.com/Azure/azure-c-shared-utility/blob/master/adapters/x509_schannel.c

Addresses https://github.com/awslabs/aws-c-io/issues/261 and pulls in https://github.com/awslabs/aws-c-io/pull/259

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/windows/windows_pki_utils.c [hunk 1]
```diff
@@ -181,31 +198,269 @@ int aws_import_trusted_certificates(
...
+    DWORD public_key_blob_length = public_key_blob->cbData;
+    if (public_key_blob_length == 0) {
+        AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: invalid zero-length ecc key data");
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        goto done;
+    }
+
+    public_key_blob_length--;
+    struct aws_byte_cursor public_blob_cursor = {
+        .ptr = public_key_blob->pbData + 1,
```
Comment (justinboswell) [inline source/windows/windows_pki_utils.c:350]: Why is this +1? Could you add a comment?

File: source/windows/windows_pki_utils.c [hunk 2]
```diff
@@ -181,31 +198,269 @@ int aws_import_trusted_certificates(
...
+
+    CRYPT_BIT_BLOB *public_key_blob = &cert_context->pCertInfo->SubjectPublicKeyInfo.PublicKey;
+    DWORD public_key_blob_length = public_key_blob->cbData;
+    if (public_key_blob_length == 0) {
+        AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: invalid zero-length ecc key data");
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        goto done;
+    }
+
+    public_key_blob_length--;
```
Comment (justinboswell) [inline source/windows/windows_pki_utils.c:348]: Is this ignoring a null terminator?

File: source/windows/windows_pki_utils.c (hunk 1 above)
Comment (bretambrose) [reply source/windows/windows_pki_utils.c:380]: Now with a big ol' comment.  Not completely resolved, but I'm not going to try and manually edit the DER form of a certificate to test a hypothesis that is mostly of curiosity.

File: source/windows/windows_pki_utils.c (hunk 2 above)
Comment (bretambrose) [reply source/windows/windows_pki_utils.c:378]: per above

**Review by justinboswell — APPROVED** [review]
I asked the question, but I wasn't ready for the answer :( Nonetheless, let it fly!

**Pull request merged**